### PR TITLE
[Bug] data is not consistent after server disconnect

### PR DESCRIPTION
**Pull request by hiveer**
1. Combine new user and user refresh together
  when server is disconnected, the first client which refresh the page will lost all data except it self
So, when the second client send a user_refresh to server, it will assume other client are working well. Obviously, this is not correct. Now I remove user_refresh logic, so each reconnection will assume it is new to server.
2. Also reconnect when server is disconnected

### CODE REVIEW

**Comment by just4fun** [comment]
I went through whole project this afternoon, what is `new_user_sync` or `new_connection_sync` used for? When there is new user, all clients will receive the message and we will invoke `append_user`, is that not enough?

However, the bug you described is existed as I verified in my localhost.

I think this PR should be CRed once all of us have a better understanding on the logics.

**Comment by hiveer** [comment]
First I agree with your conclusion in your last sentence. Then below are the answers to your two questions:
1. The difference between `new_user_sync` and `new_connection_sync`
  `new_user_sync` is a message type in the first edition of this poker, which means when a new user send me a message which tell me he/she is a new user, then I need to tell him/her who I am. So I need to send back a message and the type of message call `new_user_sync`.
  there has another message type called `user_refresh` in first edition, which will be used when a user refresh his/her browser, I found there already has user info stored in cookie store. So I treat the action as a refresh and tell other clients send back their info.
  But the logic above has a hidden bug, as the PR's description shows.

So now, I combine `new_user_sync` and `user_refresh` together to `new_connection_sync`. That means no matter you are the first time connected to the server or not. I will treat you as the new user and sync you to other clients, also sync others to you.

2. For the second question, what you described is just sync new user to all clients. But we also need to sync all clients to new user.

File: scripts/session.js [hunk 1]
```diff
@@ -140,16 +136,17 @@ ws.onmessage = function(evt){
...
       this.className = '';
     });
   }
 };
 
-// add event handler for diconnection
+// reconnect when server is disconnected
 ws.onclose= function(evt){
-  alert('Server Diconnected');
+  window.location.reload();
```
Comment (hyjk2000) [inline scripts/session.js]: We may look for a better way to handle auto reconnect. See also: https://github.com/joewalnes/reconnecting-websocket

**Pull request merged**